hikari/errors.py

```diff
@@ -90,6 +91,44 @@ class HikariWarning(RuntimeWarning):
     """
 
 
+@attr.define(auto_attribs=False, auto_exc=True, repr=False, weakref_slot=False)
+class MultiError(HikariError):
```

Comment:
@davfsa not sure if you'll be a fan of this but I'd rather keep the error more generic rn rather than case specific

Reply:
I mean it does show the error raised? They're all in the repr

Reply:
And what if they catch and handle the error? You've now just spammed their console with a ton of rubbish theyd handled

Reply:
You already do this on delete messages btw

Reply:
The gateway behaviour is just an artifact of the fact that its done through an event dispatch system, its not the best way to handle starting + stopping callbacks and if a starting callback fails there's no guarantee the server will have setup all the resources it needs to function and the assumption should be that it hasnt

This is modelled after asgi which would call a set of startup failed callbacks after a failed startup instead of the shutdown callbacks as the shutdown callbacks would only be called if server had been well successfully started

Reply:
The approach of just throwing the error to asyncio would make it a lot harder to actually handle said errors while associating them as having killed the application

Reply:
As the example I showed before goes it is possibly to concatenate traceback but I'm not sure you'd like that either lol https://github.com/python-trio/trio/blob/7b86d20549bba2629eb929a72022a19b999c91d5/trio/_core/_multierror.py

Reply:
Bump to this :)

I would rather just logging the errors instead of actually exiting with an error for consistency with GatewayBot